gobblin-runtime/src/main/java/org/apache/gobblin/runtime/api/FsSpecConsumer.java

```diff
@@ -112,8 +113,11 @@ public FsSpecConsumer(@Nullable FileSystem fs, Config config) {
         JobSpec.Builder jobSpecBuilder = new JobSpec.Builder(avroJobSpec.getUri());
         Properties props = new Properties();
         props.putAll(avroJobSpec.getProperties());
-        jobSpecBuilder.withJobCatalogURI(avroJobSpec.getUri()).withVersion(avroJobSpec.getVersion())
-            .withDescription(avroJobSpec.getDescription()).withConfigAsProperties(props);
+        jobSpecBuilder.withJobCatalogURI(avroJobSpec.getUri())
+            .withVersion(avroJobSpec.getVersion())
+            .withDescription(avroJobSpec.getDescription())
+            .withConfigAsProperties(props)
```

Comment:
Do we still need this line?

Reply:
Yes, we will still need this line. The only change is we are now explicit about how the config member variable of JobSpec is set. Previously, it defaults to using ConfigUtils#propertiesToTypedConfig() which throws an exception when it encounters prefixed keys.

Reply:
This API is quite confusing ... But I get the point why both will be needed.